smartsim/_core/control/controller.py

```diff
@@ -113,8 +119,9 @@ def start(
         execution of all jobs.
         """
         self._jobs.kill_on_interrupt = kill_on_interrupt
+
         # register custom signal handler for ^C (SIGINT)
-        signal.signal(signal.SIGINT, self._jobs.signal_interrupt)
+        SignalInterceptionStack.push(signal.SIGINT, self._jobs.signal_interrupt)
```

Comment:
Alternatively we could check to see if the signal_interrupt is already in the list of SignalInterceptionStack callback lists before adding.

Reply:
Good point! Per offline discussion, decided that we didn't want to exposed the `_callbacks` attr to users of the class, but did decided to have the controller know to call a new `push_unique` method